src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Filter';
+import { removeTodo } from '../../api/todos';
+import { ErrorType } from '../../types/Error';
+
+type Props = {
+  todos: Todo[];
+  filter: Filter;
+  setFilter: React.Dispatch<React.SetStateAction<Filter>>;
+  setError: React.Dispatch<React.SetStateAction<ErrorType>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setProcessings: React.Dispatch<React.SetStateAction<number[]>>;
+};
+
+export const Footer: FC<Props> = ({
+  todos,
+  filter,
+  setFilter,
+  setError,
+  setTodos,
+  setProcessings,
+}) => {
+  const handleOnClickFilter = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    setFilter(event.currentTarget.innerText as Filter);
+  };
+
+  const filterNames = Object.values(Filter);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const handleClearActive = () => {
```

Comment:
The function handleClearActive should be renamed to handleClearCompleted, as it's purpose is to clear completed todos.